auth/src/main/java/org/openedx/auth/presentation/signup/SignUpViewModel.kt

```diff
@@ -69,35 +74,75 @@ class SignUpViewModel(
         _uiState.update { it.copy(isLoading = true) }
         viewModelScope.launch {
             try {
-                val allFields = interactor.getRegistrationFields()
-                _uiState.update { state ->
-                    state.copy(
-                        allFields = allFields,
-                        isLoading = false,
-                    )
-                }
+                updateFields(interactor.getRegistrationFields())
             } catch (e: Exception) {
                 if (e.isInternetError()) {
                     _uiMessage.emit(
                         UIMessage.SnackBarMessage(
-                            resourceManager.getString(R.string.core_error_no_connection)
+                            resourceManager.getString(coreR.string.core_error_no_connection)
                         )
                     )
                 } else {
                     _uiMessage.emit(
                         UIMessage.SnackBarMessage(
-                            resourceManager.getString(R.string.core_error_unknown_error)
+                            resourceManager.getString(coreR.string.core_error_unknown_error)
                         )
                     )
                 }
+            } finally {
+                _uiState.update { state ->
+                    state.copy(isLoading = false)
+                }
             }
         }
     }
 
+    private fun updateFields(allFields: List<RegistrationField>) {
+        val mutableAllFields = allFields.toMutableList()
+        val requiredFields = mutableListOf<RegistrationField>()
+        val optionalFields = mutableListOf<RegistrationField>()
+        val agreementFields = mutableListOf<RegistrationField>()
+        val agreementText = agreementProvider.getAgreement(isSignIn = false)
```

Comment:
we can simply use `agreementText` instead of converting it into a Registration Field.

Reply:
this code is a part of the OpenEdx project, so it should have a maximum compatibility with all OpenEdx instances, not only Edx

Reply:
we can use an extension method to convert String into RegistrationField.
```
fun String.toRegistrationField(name: String): RegistrationField {
    return RegistrationField(
        name = name,
        label = this,
        type = RegistrationFieldType.PLAINTEXT,
        placeholder = "",
        instructions = "",
        exposed = false,
        required = false,
        restrictions = RegistrationField.Restrictions(),
        options = emptyList(),
        errorInstructions = ""
    )
}